pkg/registry/core/shoot/storage/tableconvertor.go

```diff
@@ -107,6 +114,19 @@ func (c *convertor) ConvertToTable(ctx context.Context, obj runtime.Object, tabl
 		} else {
 			cells = append(cells, "<pending>")
 		}
+		status, ok := shoot.Labels[v1beta1constants.ShootStatus]
+		if !ok {
+			cells = append(cells, "<pending>")
+		} else {
+			cells = append(cells, status)
+		}
+
+		if purpose := shoot.Spec.Purpose; purpose != nil {
+			cells = append(cells, string(*purpose))
+		} else {
+			cells = append(cells, "<none>")
+		}
+		cells = append(cells, shoot.Status.Gardener.Version)
```

Comment:
This field can be empty for a Shoot that fails to be scheduled for example.

```
$ k get shoot -A -o wide
NAMESPACE        NAME          CLOUDPROFILE   PROVIDER   REGION       SEED     K8S VERSION   HIBERNATION   LAST OPERATION               STATUS      PURPOSE   GARDENER VERSION   APISERVER   CONTROL     NODES       SYSTEM      AGE
garden-foooooo   azure-local   azure          azure      westeurope   <none>   1.21.0        Awake         <pending>                    <pending>   testing                      <unknown>   <unknown>   <unknown>   <unknown>   28s
```

You could add a handling for it and show for example `<unknown>`.

Reply:
Nice catch, thanks!